CMakeLists.txt

```diff
@@ -322,13 +322,14 @@ endfunction()
 # We migth be using the wrong version of a command, so record them all.
 function(print_versions)
   find_version("cmake" "--version" TRUE)
-  find_version(${CMAKE_COMMAND} "--version" FALSE)
 
-  if(CMAKE_MAKE_PROGRAM MATCHES "ninja")
-    find_version("ninja" "--version" TRUE)
-    find_version(${CMAKE_MAKE_PROGRAM} "--version" FALSE)
-  elseif(CMAKE_MAKE_PROGRAM MATCHES "make")
-    find_version("make" "--version" TRUE)
+  message(STATUS "Finding version for: ${CMAKE_COMMAND}")
+  message(STATUS "Found version: ${CMAKE_VERSION}")
+  message(STATUS "")
+
+  if(CMAKE_MAKE_PROGRAM STREQUAL "ninja" OR CMAKE_MAKE_PROGRAM STREQUAL "make")
+    get_filename_component(CMAKE_MAKE_PROGRAM_BN "${CMAKE_MAKE_PROGRAM}" NAME_WE)
```

Comment:
If the condition is `true`, `CMAKE_MAKE_PROGRAM_BN` is always the same as `CMAKE_MAKE_PROGRAM`, isn't it?